frame/proxy/src/lib.rs

```diff
@@ -137,84 +89,84 @@ pub struct Announcement<AccountId, Hash, BlockNumber> {
-		TooMany,
-		/// Proxy registration not found.
-		NotFound,
-		/// Sender is not a proxy of the account to be proxied.
-		NotProxy,
-		/// A call which is incompatible with the proxy type's filter was attempted.
-		Unproxyable,
-		/// Account is already a proxy.
-		Duplicate,
-		/// Call may not be made by proxy because it may escalate its privileges.
-		NoPermission,
-		/// Announcement, if made at all, was made too recently.
-		Unannounced,
-		/// Cannot add self as proxy.
-		NoSelfProxy,
-	}
-}
+	/// Configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-decl_event! {
-	/// Events type.
-	pub enum Event<T> where
-		AccountId = <T as frame_system::Config>::AccountId,
-		ProxyType = <T as Config>::ProxyType,
-		Hash = CallHashOf<T>,
-	{
-		/// A proxy was executed correctly, with the given \[result\].
-		ProxyExecuted(DispatchResult),
-		/// Anonymous account has been created by new proxy with given
-		/// disambiguation index and proxy type. \[anonymous, who, proxy_type, disambiguation_index\]
-		AnonymousCreated(AccountId, AccountId, ProxyType, u16),
-		/// An announcement was placed to make a call in the future. \[real, proxy, call_hash\]
-		Announced(AccountId, AccountId, Hash),
-	}
-}
+		/// The overarching call type.
+		type Call: Parameter + Dispatchable<Origin=Self::Origin, PostInfo=PostDispatchInfo>
+			+ GetDispatchInfo + From<frame_system::Call<Self>> + IsSubType<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::Call>;
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		type Error = Error<T>;
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
 
-		/// Deposit one of this module's events by using the default implementation.
-		fn deposit_event() = default;
+		/// A kind of proxy; specified with the proxy and passed in to the `IsProxyable` fitler.
+		/// The instance filter determines whether a given call may be proxied under this type.
+		///
+		/// IMPORTANT: `Default` must be provided and MUST BE the the *most permissive* value.
+		type ProxyType: Parameter + Member + Ord + PartialOrd + InstanceFilter<<Self as Config>::Call>
+			+ Default;
 
 		/// The base amount of currency needed to reserve for creating a proxy.
-		const ProxyDepositBase: BalanceOf<T> = T::ProxyDepositBase::get();
+		///
+		/// This is held for an additional storage item whose value size is
+		/// `sizeof(Balance)` bytes and whose key size is `sizeof(AccountId)` bytes.
+		#[pallet::constant]
+		type ProxyDepositBase: Get<BalanceOf<Self>>;
 
 		/// The amount of currency needed per proxy added.
-		const ProxyDepositFactor: BalanceOf<T> = T::ProxyDepositFactor::get();
+		///
+		/// This is held for adding 32 bytes plus an instance of `ProxyType` more into a pre-existing
+		/// storage value.
```

Comment:
I don't understand what this doc string says; can you rephrase it a bit you think?

Reply:
it is saying that we should modify `ProxyDepositFactor` to take into account 32 bytes of data + `proxytype.encode().len()` more bytes of data.

It is a hint for the person configuring this pallet to pick a reasonable number for a deposit.

Reply:
@shawntabrizi do you think it is worth updating the comment or should it already be clear enough to the target audience?

Heres a sample of how it could be updated, but not sure it really adds much.
```rust
/// This is held for adding 32 bytes plus an instance of `ProxyType` more into a pre-existing
/// storage value. Thus, when configuring `ProxyDepositFactor` one should take into account 
/// `32 + proxytype.encode().len()` bytes of data.
```

Reply:
yeah that looks like a reasonable addition :)

Couldn't hurt i think.